rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1669,6 +1665,22 @@ public BooleanWrapper shutdown(boolean preempt) {
                 removeAllNodes(entry.getKey(), preempt, true);
                 entry.getValue().shutdown(caller);
             }
+
+            boolean atLeastOneAlive = false;
+            int millisBeforeHardShutdown = 0;
+            do {
+                millisBeforeHardShutdown++;
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

Reply:
you should extract the try catch block outside the do/while

Reply:
Done.